src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java

```diff
@@ -90,10 +90,21 @@ public TempBlob removeMirrorUrlFromArtifactsXml(final TempBlob artifact,
           try (OutputStream xmlOut = xmlOutputStream(extension, tempFile)) {
             XMLInputFactory inputFactory = XMLInputFactory.newFactory();
             XMLOutputFactory outputFactory = XMLOutputFactory.newFactory();
-            XMLEventReader reader = inputFactory.createXMLEventReader(xmlIn);
-            XMLEventWriter writer = outputFactory.createXMLEventWriter(xmlOut);
-
-            streamXmlToWriterAndRemoveAbsoluteUrls(reader, writer);
+            XMLEventReader reader = null;
+            XMLEventWriter writer = null;
+            try {
```

Comment:
I believe you could use a try-with resource block and simplify this.

Reply:
`XMLEventReader` and `XMLEventWriter` are not `AutoCloseable` for some reason

Reply:
They are intentionally not `AutoCloseable` from what I can tell.